app/[locale]/[[...path]]/page.tsx

```diff
@@ -67,7 +67,9 @@ const getPage: FC<DynamicParams> = async ({ params }) => {
     // Forces the current locale to be the Default Locale
     unstable_setRequestLocale(defaultLocale.code);
 
-    return notFound();
+    // Redirect to the default locale path
+    const pathname = dynamicRouter.getPathname(path);
+    return redirect(`/${defaultLocale.code}/${pathname}`);
```

Comment:
Would this redirect work if the website is deployed on subpaths?

What would happen if you simply remove the redirect and do something like:

```tsx
const selectedLocale = availableLocaleCodes.includes(locale) ? locale : defaultLocale.code;

unstable_setRequestLocale(selectedLocale );
```

(And replace all other usages of `locale` to `selectedLocale`?)

without any redirect or return statement. Would an English version of said page be rendered with the links working? And if the page also doesn't exist on English, would it return

Reply:
i've tried out your code idea, all not found pages now are redirected to /en counterparts but that doesn't fix the navBar `Download` link

Reply:
> Would this redirect work if the website is deployed on subpaths?

subpaths mean `https://nodejs.org/en/download`, `https://nodejs.org/en/blog` or something? I checked the following links.

- https://nodejs.org/en/download => 200 OK
- https://nodejs.org/en/blog => 200 OK
- https://nodejs.org/ja/download => Redirect to https://nodejs.org/en/download
- https://nodejs.org/ja/blog => Redirect to https://nodejs.org/en/blog
- https://nodejs.org/ja/foobarbaz => Redirect to https://nodejs.org/en/foobarbaz and 404
- https://nodejs.org/ja => Redirect to https://nodejs.org/en/

I believe it works well.

> What would happen if you simply remove the redirect and do something like ~ (And replace all other usages of locale to selectedLocale?) And if the page also doesn't exist on English, would it return a 404 then?

- https://nodejs.org/en/download => 200 OK
- https://nodejs.org/en/blog => 200 OK
- https://nodejs.org/ja/download => 200 OK but en page is shown.
- https://nodejs.org/ja/blog => 200 OK but en page is shown.
- https://nodejs.org/ja/foobarbaz => 404
- https://nodejs.org/ja/ => 500 unsupported locale error.

I don't feel this solution is good idea.

 if `/ja` pages could be treated as correct paths, it would be better to change i18n/config.json  like:

```
  {
    "code": "ja",
    "localName": "日本語",
    "name": "Japanese",
    "langDir": "ltr",
    "dateFormat": "YYYY.MM.DD",
    "hrefLang": "ja",
    "enabled": true, // previous false
    "default": false
  },
```

This change will enable Japanese page.  And links work fine.

- https://nodejs.org/en/download => 200 OK
- https://nodejs.org/en/blog => 200 OK
- https://nodejs.org/ja/download => 200 OK but en page is shown.
- https://nodejs.org/ja/blog => 200 OK but en page is shown.
- https://nodejs.org/ja/foobarbaz => 404
- https://nodejs.org/ja/ => 200 OK

IMHO, I feel redirect solution is better than locale enable solution. 

1. /ja pages does not have any translated pages. navigation bars, indexes etc...
2. other locales would be better to handle as correct paths? In that case, we need to update all locales to enable.

Reply:
So the reasoning behind not using redirect is because we should show 404 pages if a page doesn't actually exist instead of redirecting.

With the proposed approach:

- /ja/download would render the en download page, but still keep the /ja in the URL
- /ja/fsdjhfkisdhf would render a Not Found

But I'm fine with redirect tho.

Reply:
I don't think you have the data to back you up. Please avoid speculating that these 404s requests wouldn't be a huge traffic.

The actual data, from Vercel logs, show that a lot of requests are 404. Yes, if actual languages are enabled that would reduce it, but you gotta understand every extra redirect is another request we need to process. For the user on a browser it might be transparent but 🤷

Im not really convinced with this approach. I understand it is a simple fire and go approach, but within this repository we did a lot of work to ensure we have a good performance tradeoff.

I would really appreciate if you could find an alternative. At this point I believe I'd only merge this if @bmuenzenmeyer is also fine with the change.

Reply:
I don't think we prioritise performance over translations, that's an unfair statement and quite provocative 😅

What Im trying to imply here is to propel a way that'd allowed us yo have both features working without a redirect. I'm sure that should be possible.

To be honest, Im not sure if the redirects depending on how many different distinct 404s we would have would affect our Vercel infrastructure. I do believe the effect is negligible, but I do believe that there might be a solution that won't require redirects. I might as well be wrong, but I want to have a 2nd opinion from Brian, before we merge this, as this PR has a great deal of an impact after all.

Reply:
Thank you. I hope I could give one step forward to tackle this problem. Lots of pages are linked from blogs, books, tutorials in Japan. I believe our solution will solve those dead links.